## Assignment 1: Solutions
Author: Laura Bissoltan

## Problem 1

#1.1 

The force acting on the satellite is gravitational force. It keeps satellite to move in a circular path and causes centripetal force. Those two forces are equal and opposite in magnitude. Therefore, using Newton's second law it can be written $(1)$

$$ F_c = ma = \frac{mv^2}{r} = G\frac{mM}{r^2} = F_G $$    

where $m$ and $M$ are mass of the satellite and the Earth respectively, $v$ is a velocity, $a$ is an acceleration in circular motion and $r$ is a distance from the center of the Earth to the center of the satellite. The velocity is defined as $v=distance/time$, where distance can be defined as a circumference of circle and time is period T: 

$$ v = \frac{2\pi r}{T} $$

By making substitution of v in (1) with this velocity above and solving for r gives:

$$\frac{4\pi^2 r^2}{T^2} = \frac{GM}{r}$$

$$ r = (\frac{GMT^2}{4\pi^2})^{1/3} $$

Since $r$ is the distance from the center of Earth to the center of satellite, the altitude $h$ above the Earth’s surface that the satellite should have is

$$ h = (\frac{GMT^2}{4\pi^2})^{1/3} - R $$

where $R$ is radius of the Earth. 


In [1]:
#1.2
from math import pi

R = 6371000 #m
G = 6.67*10** -(11) #m^3*kg^(-1)s^(-2)
M = 5.97*10**24 #kg
T = float(input("Enter the value of T in seconds:" ))
h = ((G*M*T**2)/(4*pi**2))**(1/3) - R
print("Altitude is: ", h, "meters")


Enter the value of T in seconds:5400
Altitude is:  279321.62537285965 meters


#1.3

Table 1. The altitudes of satellites    
    
|      Period(T)    |  Period(T)/seconds  |   Altitude(h)/meters |
|:-----------------:|:-------------------:|:--------------------:|
| once a day        |      86400          |     35855910.1762    |
| once every 90 min |      5400           |     279321.6254      |
| once every 45 min |      2700           |    -2181559.8978     |

The value of altitude is negative for 45 min and it can be concluded that the altitude of satellite which orbit the Earth once every 45 min is not possible.


#1.4

The sidereal day (23.93 hours) is one complete rotation of the Earth about its axis with respect to the distant stars. It is shorter than solar day (24 hours) because, in addition to the rotation, the Earth has to move slightly more along the path around Sun in order to complete rotation with respect to the Sun.
Using the program above altitude was calculated for 23.93 hours and it was 35773762.3298 meters. The difference was 82147.8463 meters. 

## Problem 2

In [2]:
#2.1

n_list=[]

def multinomial_coeff(n_list):
    val = 1
    i = 1
    for j in n_list:
        for k in range(1,j+1):
            val = val * i
            val //= k
            i = i + 1
    return val

n = int(input("Enter the number of elements: ")) 

for i in range(0, n): 
    print("Enter a nonnegative integer: ")
    elt = int(input()) 
  
    n_list.append(elt)
    
print(multinomial_coeff(n_list))

Enter the number of elements: 6
Enter a nonnegative integer: 
3
Enter a nonnegative integer: 
3
Enter a nonnegative integer: 
0
Enter a nonnegative integer: 
1
Enter a nonnegative integer: 
1
Enter a nonnegative integer: 
1
10080


In [3]:
#2.2

print("a) ", multinomial_coeff([4, 2, 2, 0]))
print("b) ", multinomial_coeff([3, 3, 0, 1, 1, 1]))
print("c) ", multinomial_coeff([2, 3, 2, 4, 1, 0]))

a)  420
b)  10080
c)  831600


In [4]:
#2.3

#since by formula order does not matter in calculation of the sum, some number combinations was multiplied by 2, 3 or 6  
#e.g. multinomial_coeff([0,0,5]) = multinomial_coeff([0,5,0]) = multinomial_coeff([5,0,0]) = 1 (multiplied by 3 because permutation 3!/(2!1!)=3) 

a_sum = (multinomial_coeff([0,6]) + multinomial_coeff([1,5]) + multinomial_coeff([2,4]))*2 + multinomial_coeff([3,3])
b_sum = (multinomial_coeff([0,0,5]) + multinomial_coeff([1, 1, 3])+multinomial_coeff([2,2,1]))*3 + (multinomial_coeff([0, 1, 4]) + multinomial_coeff([0,2,3]))*6

print("a) Sum of the all possible multinomial coefficients for r=2 n=6 is", a_sum)
print("b) Sum of the all possible multinomial coefficients for r=3 n=5 is", b_sum)



a) Sum of the all possible multinomial coefficients for r=2 n=6 is 64
b) Sum of the all possible multinomial coefficients for r=3 n=5 is 243


## Problem 3

In [5]:
#3.1

def catalan(k):
    if k <= 1:
        return 1
    else:
        return (4*k-2)*catalan(k-1)//(k+1)
    

In [6]:
#3.2

print("The value of C_19 is", catalan(19))
print("The value of C_99 is", catalan(99))

The value of C_19 is 1767263190
The value of C_99 is 227508830794229349661819540395688853956041682601541047340
